File: python/swagger_client/models/vault_credit_card.py
# coding: utf-8

class VaultCreditCard(object):
    swagger_types = {
        'magnetic_tracks': 'str',
        'track1': 'str',
        'track2': 'str',
        'number': 'str',
        'masked_number': 'str',
        'holder': 'str',
        'exp_date': 'str',
        'cvv': 'str',
        'address': 'str',
        'city': 'str',
        'state': 'str',
        'zip': 'str',
        'swiped': 'bool',
        'type_string': 'str',
        'cvv_type': 'CVVPresenceType',
        'not_present': 'bool',
        'response': 'Response',
        'card_id': 'int',
        'note': 'str',
        'main': 'bool'
    }

    attribute_map = {
        'magnetic_tracks': 'MagneticTracks',
        'track1': 'Track1',
        'track2': 'Track2',
        'number': 'Number',
        'masked_number': 'MaskedNumber',
        'holder': 'Holder',
        'exp_date': 'ExpDate',
        'cvv': 'CVV',
        'address': 'Address',
        'city': 'City',
        'state': 'State',
        'zip': 'ZIP',
        'swiped': 'Swiped',
        'type_string': 'TypeString',
        'cvv_type': 'CVVType',
        'not_present': 'NotPresent',
        'response': 'Response',
        'card_id': 'CardID',
        'note': 'Note',
        'main': 'Main'
    }

    def __init__(self, magnetic_tracks=None, track1=None, track2=None, number=None, masked_number=None, holder=None, exp_date=None, cvv=None, address=None, city=None, state=None, zip=None, swiped=None, type_string=None, cvv_type=None, not_present=None, response=None, card_id=None, note=None, main=None):  # noqa: E501
        """VaultCreditCard - a model defined in Swagger"""  # noqa: E501
        self._magnetic_tracks = None
        self._track1 = None
        self._track2 = None
        self._number = None
        self._masked_number = None
        self._holder = None
        self._exp_date = None
        self._cvv = None
        self._address = None
        self._city = None
        self._state = None
        self._zip = None
        self._swiped = None
        self._type_string = None
        self._cvv_type = None
        self._not_present = None
        self._response = None
        self._card_id = None
        self._note = None
        self._main = None
        self.discriminator = None
        if magnetic_tracks is not None:
            self.magnetic_tracks = magnetic_tracks
        if track1 is not None:
            self.track1 = track1
        if track2 is not None:
            self.track2 = track2
        if number is not None:
            self.number = number
        if masked_number is not None:
            self.masked_number = masked_number
        if holder is not None:
            self.holder = holder
        if exp_date is not None:
            self.exp_date = exp_date
        if cvv is not None:
            self.cvv = cvv
        if address is not None:
            self.address = address
        if city is not None:
            self.city = city
        if state is not None:
            self.state = state
        if zip is not None:
            self.zip = zip
        if swiped is not None:
            self.swiped = swiped
        if type_string is not None:
            self.type_string = type_string
        if cvv_type is not None:
            self.cvv_type = cvv_type
        if not_present is not None:
            self.not_present = not_present
        if response is not None:
            self.response = response
        if card_id is not None:
            self.card_id = card_id
        if note is not None:
            self.note = note
        if main is not None:
            self.main = main

    @property
    def magnetic_tracks(self):
        return self._magnetic_tracks

    @magnetic_tracks.setter
    def magnetic_tracks(self, magnetic_tracks):
        self._magnetic_tracks = magnetic_tracks

    @property
    def track1(self):
        return self._track1

    @track1.setter
    def track1(self, track1):
        self._track1 = track1

    @property
    def track2(self):
        return self._track2

    @track2.setter
    def track2(self, track2):
        self._track2 = track2

    @property
    def number(self):
        return self._number

    @number.setter
    def number(self, number):
        self._number = number

    @property
    def masked_number(self):
        return self._masked_number

    @masked_number.setter
    def masked_number(self, masked_number):
        self._masked_number = masked_number

    @property
    def holder(self):
        return self._holder

    @holder.setter
    def holder(self, holder):
        self._holder = holder

    @property
    def exp_date(self):
        return self._exp_date

    @exp_date.setter
    def exp_date(self, exp_date):
        self._exp_date = exp_date

    @property
    def cvv(self):
        return self._cvv

    @cvv.setter
    def cvv(self, cvv):
        self._cvv = cvv

    @property
    def address(self):
        return self._address

    @address.setter
    def address(self, address):
        self._address = address

    @property
    def city(self):
        return self._city

    @city.setter
    def city(self, city):
        self._city = city

    @property
    def state(self):
        return self._state

    @state.setter
    def state(self, state):
        self._state = state

    @property
    def zip(self):
        return self._zip

    @zip.setter
    def zip(self, zip):
        self._zip = zip

    @property
    def swiped(self):
        return self._swiped

    @swiped.setter
    def swiped(self, swiped):
        self._swiped = swiped

    @property
    def type_string(self):
        return self._type_string

    @type_string.setter
    def type_string(self, type_string):
        self._type_string = type_string

    @property
    def cvv_type(self):
        return self._cvv_type

    @cvv_type.setter
    def cvv_type(self, cvv_type):
        self._cvv_type = cvv_type

    @property
    def not_present(self):
        return self._not_present

    @not_present.setter
    def not_present(self, not_present):
        self._not_present = not_present

    @property
    def response(self):
        return self._response

    @response.setter
    def response(self, response):
        self._response = response

    @property
    def card_id(self):
        return self._card_id

    @card_id.setter
    def card_id(self, card_id):
        self._card_id = card_id

    @property
    def note(self):
        return self._note

    @note.setter
    def note(self, note):
        self._note = note

    @property
    def main(self):
        return self._main

    @main.setter
    def main(self, main):
        self._main = main

File: CodeSamples/php/SwaggerClient-php/lib/Model/VaultCreditCard.php
<?php

namespace Swagger\Client\Model;

use \ArrayAccess;
use \Swagger\Client\ObjectSerializer;

class VaultCreditCard implements ModelInterface, ArrayAccess
{
    const DISCRIMINATOR = null;

    protected static $swaggerModelName = 'VaultCreditCard';

    protected static $swaggerTypes = [
        'magnetic_tracks' => 'string',
        'track1' => 'string',
        'track2' => 'string',
        'number' => 'string',
        'masked_number' => 'string',
        'holder' => 'string',
        'exp_date' => 'string',
        'cvv' => 'string',
        'address' => 'string',
        'city' => 'string',
        'state' => 'string',
        'zip' => 'string',
        'swiped' => 'bool',
        'type_string' => 'string',
        'cvv_type' => '\Swagger\Client\Model\CVVPresenceType',
        'not_present' => 'bool',
        'response' => '\Swagger\Client\Model\Response',
        'card_id' => 'int',
        'note' => 'string',
        'main' => 'bool'
    ];

    protected static $swaggerFormats = [
        'magnetic_tracks' => null,
        'track1' => null,
        'track2' => null,
        'number' => null,
        'masked_number' => null,
        'holder' => null,
        'exp_date' => null,
        'cvv' => null,
        'address' => null,
        'city' => null,
        'state' => null,
        'zip' => null,
        'swiped' => null,
        'type_string' => null,
        'cvv_type' => null,
        'not_present' => null,
        'response' => null,
        'card_id' => 'int32',
        'note' => null,
        'main' => null
    ];

    public static function swaggerTypes()
    {
        return self::$swaggerTypes;
    }

    public static function swaggerFormats()
    {
        return self::$swaggerFormats;
    }

    protected static $attributeMap = [
        'magnetic_tracks' => 'MagneticTracks',
        'track1' => 'Track1',
        'track2' => 'Track2',
        'number' => 'Number',
        'masked_number' => 'MaskedNumber',
        'holder' => 'Holder',
        'exp_date' => 'ExpDate',
        'cvv' => 'CVV',
        'address' => 'Address',
        'city' => 'City',
        'state' => 'State',
        'zip' => 'ZIP',
        'swiped' => 'Swiped',
        'type_string' => 'TypeString',
        'cvv_type' => 'CVVType',
        'not_present' => 'NotPresent',
        'response' => 'Response',
        'card_id' => 'CardID',
        'note' => 'Note',
        'main' => 'Main'
    ];

    protected static $setters = [
        'magnetic_tracks' => 'setMagneticTracks',
        'track1' => 'setTrack1',
        'track2' => 'setTrack2',
        'number' => 'setNumber',
        'masked_number' => 'setMaskedNumber',
        'holder' => 'setHolder',
        'exp_date' => 'setExpDate',
        'cvv' => 'setCvv',
        'address' => 'setAddress',
        'city' => 'setCity',
        'state' => 'setState',
        'zip' => 'setZip',
        'swiped' => 'setSwiped',
        'type_string' => 'setTypeString',
        'cvv_type' => 'setCvvType',
        'not_present' => 'setNotPresent',
        'response' => 'setResponse',
        'card_id' => 'setCardId',
        'note' => 'setNote',
        'main' => 'setMain'
    ];

    protected static $getters = [
        'magnetic_tracks' => 'getMagneticTracks',
        'track1' => 'getTrack1',
        'track2' => 'getTrack2',
        'number' => 'getNumber',
        'masked_number' => 'getMaskedNumber',
        'holder' => 'getHolder',
        'exp_date' => 'getExpDate',
        'cvv' => 'getCvv',
        'address' => 'getAddress',
        'city' => 'getCity',
        'state' => 'getState',
        'zip' => 'getZip',
        'swiped' => 'getSwiped',
        'type_string' => 'getTypeString',
        'cvv_type' => 'getCvvType',
        'not_present' => 'getNotPresent',
        'response' => 'getResponse',
        'card_id' => 'getCardId',
        'note' => 'getNote',
        'main' => 'getMain'
    ];

    public static function attributeMap()
    {
        return self::$attributeMap;
    }

    public static function setters()
    {
        return self::$setters;
    }

    public static function getters()
    {
        return self::$getters;
    }

    public function getModelName()
    {
        return self::$swaggerModelName;
    }

    protected $container = [];

    public function __construct(array $data = null)
    {
        $this->container['magnetic_tracks'] = isset($data['magnetic_tracks']) ? $data['magnetic_tracks'] : null;
        $this->container['track1'] = isset($data['track1']) ? $data['track1'] : null;
        $this->container['track2'] = isset($data['track2']) ? $data['track2'] : null;
        $this->container['number'] = isset($data['number']) ? $data['number'] : null;
        $this->container['masked_number'] = isset($data['masked_number']) ? $data['masked_number'] : null;
        $this->container['holder'] = isset($data['holder']) ? $data['holder'] : null;
        $this->container['exp_date'] = isset($data['exp_date']) ? $data['exp_date'] : null;
        $this->container['cvv'] = isset($data['cvv']) ? $data['cvv'] : null;
        $this->container['address'] = isset($data['address']) ? $data['address'] : null;
        $this->container['city'] = isset($data['city']) ? $data['city'] : null;
        $this->container['state'] = isset($data['state']) ? $data['state'] : null;
        $this->container['zip'] = isset($data['zip']) ? $data['zip'] : null;
        $this->container['swiped'] = isset($data['swiped']) ? $data['swiped'] : null;
        $this->container['type_string'] = isset($data['type_string']) ? $data['type_string'] : null;
        $this->container['cvv_type'] = isset($data['cvv_type']) ? $data['cvv_type'] : null;
        $this->container['not_present'] = isset($data['not_present']) ? $data['not_present'] : null;
        $this->container['response'] = isset($data['response']) ? $data['response'] : null;
        $this->container['card_id'] = isset($data['card_id']) ? $data['card_id'] : null;
        $this->container['note'] = isset($data['note']) ? $data['note'] : null;
        $this->container['main'] = isset($data['main']) ? $data['main'] : null;
    }

    public function listInvalidProperties()
    {
        $invalidProperties = [];

        return $invalidProperties;
    }

    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }

    public function getMagneticTracks()
    {
        return $this->container['magnetic_tracks'];
    }

    public function setMagneticTracks($magnetic_tracks)
    {
        $this->container['magnetic_tracks'] = $magnetic_tracks;

        return $this;
    }

    public function getTrack1()
    {
        return $this->container['track1'];
    }

    public function setTrack1($track1)
    {
        $this->container['track1'] = $track1;

        return $this;
    }

    public function getTrack2()
    {
        return $this->container['track2'];
    }

    public function setTrack2($track2)
    {
        $this->container['track2'] = $track2;

        return $this;
    }

    public function getNumber()
    {
        return $this->container['number'];
    }

    public function setNumber($number)
    {
        $this->container['number'] = $number;

        return $this;
    }

    public function getMaskedNumber()
    {
        return $this->container['masked_number'];
    }

    public function setMaskedNumber($masked_number)
    {
        $this->container['masked_number'] = $masked_number;

        return $this;
    }

    public function getHolder()
    {
        return $this->container['holder'];
    }

    public function setHolder($holder)
    {
        $this->container['holder'] = $holder;

        return $this;
    }

    public function getExpDate()
    {
        return $this->container['exp_date'];
    }

    public function setExpDate($exp_date)
    {
        $this->container['exp_date'] = $exp_date;

        return $this;
    }

    public function getCvv()
    {
        return $this->container['cvv'];
    }

    public function setCvv($cvv)
    {
        $this->container['cvv'] = $cvv;

        return $this;
    }

    public function getAddress()
    {
        return $this->container['address'];
    }

    public function setAddress($address)
    {
        $this->container['address'] = $address;

        return $this;
    }

    public function getCity()
    {
        return $this->container['city'];
    }

    public function setCity($city)
    {
        $this->container['city'] = $city;

        return $this;
    }

    public function getState()
    {
        return $this->container['state'];
    }

    public function setState($state)
    {
        $this->container['state'] = $state;

        return $this;
    }

    public function getZip()
    {
        return $this->container['zip'];
    }

    public function setZip($zip)
    {
        $this->container['zip'] = $zip;

        return $this;
    }

    public function getSwiped()
    {
        return $this->container['swiped'];
    }

    public function setSwiped($swiped)
    {
        $this->container['swiped'] = $swiped;

        return $this;
    }

    public function getTypeString()
    {
        return $this->container['type_string'];
    }

    public function setTypeString($type_string)
    {
        $this->container['type_string'] = $type_string;

        return $this;
    }

    public function getCvvType()
    {
        return $this->container['cvv_type'];
    }

    public function setCvvType($cvv_type)
    {
        $this->container['cvv_type'] = $cvv_type;

        return $this;
    }

    public function getNotPresent()
    {
        return $this->container['not_present'];
    }

    public function setNotPresent($not_present)
    {
        $this->container['not_present'] = $not_present;

        return $this;
    }

    public function getResponse()
    {
        return $this->container['response'];
    }

    public function setResponse($response)
    {
        $this->container['response'] = $response;

        return $this;
    }

    public function getCardId()
    {
        return $this->container['card_id'];
    }

    public function setCardId($card_id)
    {
        $this->container['card_id'] = $card_id;

        return $this;
    }

    public function getNote()
    {
        return $this->container['note'];
    }

    public function setNote($note)
    {
        $this->container['note'] = $note;

        return $this;
    }

    public function getMain()
    {
        return $this->container['main'];
    }

    public function setMain($main)
    {
        $this->container['main'] = $main;

        return $this;
    }
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    public function offsetGet($offset)
    {
        return isset($this->container[$offset]) ? $this->container[$offset] : null;
    }

    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }

    public function __toString()
    {
        if (defined('JSON_PRETTY_PRINT')) { // use JSON pretty print
            return json_encode(
                ObjectSerializer::sanitizeForSerialization($this),
                JSON_PRETTY_PRINT
            );
        }

        return json_encode(ObjectSerializer::sanitizeForSerialization($this));
    }
}
